src/uu/uniq/src/uniq.rs

```diff
@@ -58,22 +63,26 @@ impl Uniq {
     ) {
         let mut lines: Vec<String> = vec![];
         let mut first_line_printed = false;
-        let delimiters = &self.delimiters;
+        let delimiters = self.delimiters;
         let line_terminator = self.get_line_terminator();
+        let print_delimiters = delimiters != Delimiters::None;
```

Comment:
can you please document this a bit please? thanks

Reply:
I have updated and added comments as requested. Would have preferred to write the code to be more readable without needing comments. But I will be creating another PR refactoring this part to optimize the memory usage.